server/src/test/java/io/crate/expression/scalar/ParseURLFunctionTest.java

```diff
@@ -0,0 +1,125 @@
+        assertEvaluate(String.format("parse_url('%s')", uri), value);
+    }
+
+    @Test
+    public void test_parse_url_userinfo() {
+        String uri = "https://user:pwd@crate.io:443/";
+        Map<String, Object> value = MapBuilder.<String, Object>newMapBuilder()
+            .put("scheme", "https")
+            .put("userinfo", "user:pwd")
+            .put("hostname", "crate.io")
+            .put("port", 443)
+            .put("path", "/")
+            .put("query", null)
+            .put("parameters", null)
+            .put("fragment", null)
+            .map();
+        assertEvaluate(String.format("parse_url('%s')", uri), value);
+    }
+
+    @Test
+    public void test_parse_url_query() {
+        String uri = "https://crate.io/?foo=bar&foo=bar2&foo2=bar&foo2";
+        Map<String, Object> value = MapBuilder.<String, Object>newMapBuilder()
+            .put("scheme", "https")
+            .put("userinfo",null)
+            .put("hostname", "crate.io")
+            .put("port", null)
+            .put("path", "/")
+            .put("query", "foo=bar&foo=bar2&foo2=bar&foo2")
+            .put("parameters", MapBuilder.<String,Object>newMapBuilder()
+                                .put("foo",List.of("bar","bar2"))
+                                .put("foo2",Stream.of("bar",null).toList())
+                                .map())
+            .put("fragment", null)
+            .map();
+        assertEvaluate(String.format("parse_url('%s')", uri), value);
+    }
+
+    @Test
+    public void test_parse_url_unsafe_character() {
+        String uri = "https://crate.io/sub%20space/hello.gif";
+        Map<String, Object> value = MapBuilder.<String, Object>newMapBuilder()
+            .put("scheme", "https")
+            .put("userinfo",null)
+            .put("hostname", "crate.io")
+            .put("port", null)
+            .put("path", "/sub space/hello.gif")
+            .put("query", null)
+            .put("parameters", null)
+            .put("fragment", null)
+            .map();
+        assertEvaluate(String.format("parse_url('%s')", uri), value);
+    }
+
+    @Test
+    public void test_parse_url_complete_example() {
+        String uri = "https://user:pw%26@testing.crate.io:4200/sub+space/sub%20space2/index.html?foo=bar&foo=&foo2=https%3A%2F%2Fcrate.io%2F%3Ffoo%3Dbar%26foo%3Dbar2%26foo2#ref";
+        Map<String, Object> value = MapBuilder.<String, Object>newMapBuilder()
+            .put("scheme", "https")
+            .put("userinfo", "user:pw&")
+            .put("hostname", "testing.crate.io")
+            .put("port", 4200)
+            .put("path", "/sub space/sub space2/index.html")
+            .put("query","foo=bar&foo=&foo2=https://crate.io/?foo=bar&foo=bar2&foo2")
+            .put("parameters", MapBuilder.<String,Object>newMapBuilder()
+                                .put("foo",Stream.of("bar",null).toList())
+                                .put("foo2",List.of("https://crate.io/?foo=bar&foo=bar2&foo2"))
```

Comment:
Parameter containing `&` is an interesting edge case - maybe worth adding such example to the docs?
Let's even make it harder and and a third `key=val` after `param_containing_amp` (here and in docs)

Reply:
The `&` in the parameter needs to be encoded. The is standard URL behaviour and not specific to CrateDB. The `query` string just holds the decoded query string, which is in the documentation.

Reply:
Ah ok, so you are saying we won't have ambiguous `split("&"`).

Reply:
correct. An ampersand in a value has to be escaped as `%26`, otherwise it is separating to keys.